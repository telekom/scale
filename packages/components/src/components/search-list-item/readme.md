# scale-search-list-item



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute     | Description                                           | Type                  | Default   |
| ------------- | ------------- | ----------------------------------------------------- | --------------------- | --------- |
| `dismissible` | `dismissible` | (optional) is close button to be shown                | `boolean`             | `true`    |
| `variant`     | `variant`     | (optional) The buttons to be shown on Hover or always | `"always" \| "hover"` | `'hover'` |


## Shadow Parts

| Part                  | Description |
| --------------------- | ----------- |
| `"base"`              |             |
| `"clear-icon"`        |             |
| `"clear-icon-button"` |             |
| `"label"`             |             |
| `"prefix"`            |             |
| `"suffix"`            |             |
| `"supporting-text"`   |             |
| `"text"`              |             |


## Dependencies

### Depends on

- [scale-icon-button](../icon-button)
- [scale-icon-action-close](../icons/action-close)

### Graph
```mermaid
graph TD;
  scale-search-list-item --> scale-icon-button
  scale-search-list-item --> scale-icon-action-close
  style scale-search-list-item fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
