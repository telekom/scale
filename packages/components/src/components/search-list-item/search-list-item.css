:host {
  /* base */
  --width: 100%;
  --background-hover: var(--telekom-color-ui-state-fill-hovered);
  --font-weight: var(--telekom-typography-font-weight-medium);
  --letter-spacing-standard: var(--telekom-typography-letter-spacing-standard);
  --font-family: inherit;

  /* supporting text */
  --text-item-color: var(--telekom-color-text-and-icon-standard);
  --font-size-item-text: 16px;
  --line-height-item-text: 100%;

  /* supporting text */
  --text-supporting-text-color: var(--telekom-color-text-and-icon-additional);
  --font-size-supporting-text: 12px;
  --line-height-supporting-text: 16px;

  /* clear-icon */
  --clear-icon-color: var(--telekom-color-text-and-icon-standard);
}

[part~='base'] {
  width: var(--width);
  font-family: var(--font-family);
  font-weight: var(--font-weight);
  letter-spacing: var(--letter-spacing-standard);
  display: flex;
  justify-content: flex-start;
  align-items: center;
  padding: 8px 8px 8px 16px;
}

[part~='base'].scale-search-list-item-hover:hover {
  background-color: var(--background-hover);
}

[part~='base'].scale-search-list-item-hover:hover
  slot[name='suffix']::slotted(*) {
  --background: transparent;
}

[part~='label'],
[part~='suffix'] {
  display: flex;
}

[part~='prefix'] {
  align-self: flex-start;
}

[part~='label'] {
  font-size: var(--font-size-item-text);
  color: var(--text-item-color);
  line-height: var(--line-height-item-text);
  padding-top: 8px;
  padding-bottom: 8px;
  text-align: left;
}

[part~='supporting-text'] {
  font-size: var(--font-size-supporting-text);
  color: var(--text-supporting-text-color);
  line-height: var(--line-height-supporting-text);
  text-align: left;
}

[part~='suffix']:last-of-type {
  margin-left: auto;
}

::slotted([slot='prefix']) {
  display: inline;
  padding-right: 8px;
  margin-top: 4px;
}

[part~='clear-icon-button'] {
  --background: transparent;
}

[part~='clear-icon'] {
  color: var(--clear-icon-color);
}
