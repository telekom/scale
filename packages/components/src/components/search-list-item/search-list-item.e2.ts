import { newE2EPage } from '@stencil/core/testing';

describe('scale-search-list-item', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent(
      '<scale-search-list-item>Item 01</scale-search-list-item>'
    );
    const element = await page.find('scale-search-list-item');
    expect(element).toHaveClass('hydrated');
  });
});
