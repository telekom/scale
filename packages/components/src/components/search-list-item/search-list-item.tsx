import { Component, Element, h, Host, Prop } from '@stencil/core';
import classNames from 'classnames';
import { emitEvent } from '../../utils/utils';

@Component({
  tag: 'scale-search-list-item',
  styleUrl: 'search-list-item.css',
  shadow: true,
})
export class SearchSelectItem {
  @Element() hostElement: HTMLElement;

  /** (optional) is close button to be shown */
  @Prop() dismissible?: boolean = true;

  /** (optional) The buttons to be shown on Hover or always */
  @Prop() variant?: 'always' | 'hover' = 'hover';

  /**
   * Handles click event for close button.
   * @param event click event
   */
  handleClick = (event: MouseEvent) => {
    event.preventDefault();
    event.stopPropagation();
    emitEvent(this, 'scaleClose', event);
  };

  /**
   * Gets clear button.
   * @returns clear icon button.
   */
  getClearIconButton() {
    return (
      <scale-icon-button
        size="medium"
        part="clear-icon-button"
        onClick={this.handleClick}
      >
        <scale-icon-action-close
          part="clear-icon"
          accessibility-title="close"
          size={24}
        />
      </scale-icon-button>
    );
  }

  render() {
    return (
      <Host>
        <div part="base" tabIndex={0} class={this.getCssClassMap()}>
          <div part="prefix">
            <slot name="prefix"></slot>
          </div>

          <div part="text">
            <div part="label">
              <slot></slot>
            </div>
            <div part="supporting-text">
              <slot name="supporting-text"></slot>
            </div>
          </div>

          <div part="suffix">
            {this.dismissible ? (
              this.getClearIconButton()
            ) : (
              <slot name="suffix"></slot>
            )}
          </div>
        </div>
      </Host>
    );
  }

  getCssClassMap() {
    const component = 'scale-search-list-item';
    return classNames(`${component}-${this.variant}`);
  }
}
