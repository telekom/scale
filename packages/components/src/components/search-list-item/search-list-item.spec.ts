import { newSpecPage } from '@stencil/core/testing';
import { SearchSelectItem } from './search-list-item';

describe('SearchSelectItem', () => {
  let page;

  beforeEach(async () => {
    page = await newSpecPage({
      components: [SearchSelectItem],
      html: `<scale-search-list-item dismissible=true>
      <scale-icon-action-search
        size="24"
        slot="prefix"
      ></scale-icon-action-search>
      Item 01
      <div slot="supporting-text">Supporting text</div>
      <scale-icon-button size="medium" slot="suffix">
        <scale-icon-navigation-external-link
          size="24"
        ></scale-icon-navigation-external-link>
      </scale-icon-button>
    </scale-search-list-item>`,
    });
  });

  it('should match snapshot', async () => {
    expect(page.root).toMatchSnapshot();
  });

  it('should reflect attributes/props being set', async () => {
    const specPage = await newSpecPage({
      components: [SearchSelectItem],
      html: `<scale-search-list-item dismissible=true>
      <scale-icon-action-search
        size="24"
        slot="prefix"
      ></scale-icon-action-search>
      Item 01
      <div slot="supporting-text">Supporting text</div>
      <scale-icon-button size="medium" slot="suffix">
        <scale-icon-navigation-external-link
          size="24"
        ></scale-icon-navigation-external-link>
      </scale-icon-button>
    </scale-search-list-item>`,
    });

    expect(specPage.rootInstance.dismissible).toBe(true);
  });
});
