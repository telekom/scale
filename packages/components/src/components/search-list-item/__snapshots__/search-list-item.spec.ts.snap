// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSelectItem should match snapshot 1`] = `
<scale-search-list-item dismissible="true">
  <mock:shadow-root>
    <div class="scale-search-list-item-hover" part="base" tabindex="0">
      <div part="prefix">
        <slot name="prefix"></slot>
      </div>
      <div part="text">
        <div part="label">
          <slot></slot>
        </div>
        <div part="supporting-text">
          <slot name="supporting-text"></slot>
        </div>
      </div>
      <div part="suffix">
        <scale-icon-button part="clear-icon-button" size="medium">
          <scale-icon-action-close accessibility-title="close" part="clear-icon" size="24"></scale-icon-action-close>
        </scale-icon-button>
      </div>
    </div>
  </mock:shadow-root>
  <scale-icon-action-search size="24" slot="prefix"></scale-icon-action-search>
  Item 01
  <div slot="supporting-text">
    Supporting text
  </div>
  <scale-icon-button size="medium" slot="suffix">
    <scale-icon-navigation-external-link size="24"></scale-icon-navigation-external-link>
  </scale-icon-button>
</scale-search-list-item>
`;
